state.go

```diff
@@ -77,6 +87,12 @@ func (s *State) GuildAdd(guild *Guild) error {
 		s.channelMap[c.ID] = c
 	}
 
+	// If this guild contains a new member slice, we must regenerate the member map so the pointers stay valid
+	if guild.Members != nil {
+		delete(s.memberMap, guild.ID)
```

Comment:
Probably no benefit to deleting it first, it'll get overwritten in createMemberMap.